File: src/modules/experiment/types/experiment.ts
export type ExperimentStatus = 'idle' | 'running' | 'paused' | 'ended';

export type EventType =
  | 'experiment_start'
  | 'experiment_pause'
  | 'experiment_resume'
  | 'experiment_end'
  | 'question_start'
  | 'question_end'
  | 'question_snapshot'
  | 'rest_start'
  | 'rest_end'
  | 'stroke_start'
  | 'stroke_point'
  | 'stroke_end'
  | 'undo'
  | 'clear';

export type EventRecord = {
  timestamp: number;
  type: EventType;
  data?: Record<string, unknown>;
};

/** 单题实验数据（核心结构） */
export type ExperimentData = {
  questionId: string;
  events: EventRecord[];
  screenSnapshot?: string;
  answerDurationMs?: number;
};

export type StrokePoint = {
  x: number;
  y: number;
  pressure: number;
  timestamp: number;
};

export type DrawingTool = 'pen' | 'eraser';

export type DrawingStroke = {
  id: string;
  points: StrokePoint[];
  color: string;
  strokeWidth: number;
  tool?: DrawingTool;
};

export type QuestionItem = {
  id: string;
  title?: string;
  content: string;
};

export type ExperimentFlow = {
  id: string;
  name: string;
  description?: string;
  question_count?: number;
  rest_break_enabled?: boolean;
  rest_break_seconds?: number;
};

export type ExperimentSession = {
  sessionId: string;
  flowId: string;
  startedAt: number | null;
  endedAt: number | null;
  status: ExperimentStatus;
  currentQuestionIndex: number;
  questions: ExperimentData[];
  strokes: Record<string, DrawingStroke[]>;
};

export type QuestionAnswerStat = {
  questionId: string;
  title?: string;
  index: number;
  durationMs: number;
};

export type ExperimentResultSummary = {
  sessionId: string;
  flowId: string;
  flowName: string;
  perQuestion: QuestionAnswerStat[];
  totalAnswerMs: number;
  averageAnswerMs: number;
};

File: backend/schemas/experiment.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class ExperimentFlowItem(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    question_count: int = 0
    rest_break_enabled: bool = True
    rest_break_seconds: int = 5


class ExperimentFlowDetailResponse(BaseModel):
    errCode: int = 0
    errMsg: str = "success"
    data: Optional[ExperimentFlowItem] = None


class ExperimentFlowListResponse(BaseModel):
    errCode: int = 0
    errMsg: str = "success"
    data: List[ExperimentFlowItem] = []


class ExperimentQuestionItem(BaseModel):
    id: str
    title: Optional[str] = None
    content: str


class ExperimentQuestionListResponse(BaseModel):
    errCode: int = 0
    errMsg: str = "success"
    data: List[ExperimentQuestionItem] = []


class ExperimentSessionSubmitRequest(BaseModel):
    sessionId: str = Field(..., max_length=64)
    flowId: str = Field(..., max_length=64)
    startedAt: Optional[int] = None
    endedAt: Optional[int] = None
    status: str = Field(default="ended", max_length=20)
    questions: List[Dict[str, Any]] = Field(default_factory=list)
    strokes: Dict[str, Any] = Field(default_factory=dict)


class ExperimentSessionSubmitResponse(BaseModel):
    errCode: int = 0
    errMsg: str = "success"
    data: dict = Field(default_factory=dict)
